### PR TITLE
Porting of MolGANAggregationLayer from TF to Torch

### PR DESCRIPTION
**Pull request by shreyasvinaya**
## Description
Added MolGANAggregationLayer, Documentation related to it and its unit tests
Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/layers.py [hunk 1]
```diff
@@ -3186,6 +3186,95 @@ def forward(
...
+class MolGANAggregationLayer(nn.Module):
+    """
+    Graph Aggregation layer used in MolGAN model.
+    MolGAN is a WGAN type model for generation of small molecules.
+    Performs aggregation on tensor resulting from convolution layers.
+    Given its simple nature it might be removed in future and moved to
+    MolGANEncoderLayer.
+
+
+    Example
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:3198]: This should be Examples I think https://numpydoc.readthedocs.io/en/latest/format.html#examples

File: deepchem/models/torch_models/layers.py [hunk 2]
```diff
@@ -3186,6 +3186,95 @@ def forward(
...
+        self.units: int = units
+        self.activation = activation
+        self.dropout_rate: float = dropout_rate
+        self.name: str = name
+
+        self.d1 = nn.Linear(self.units, self.units)
+        self.d2 = nn.Linear(self.units, self.units)
+        self.dropout_layer = nn.Dropout(dropout_rate)
+
+    def __repr__(self) -> str:
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:3252]: Can you add a docstring here?

File: deepchem/models/torch_models/layers.py [hunk 3]
```diff
@@ -3097,7 +3097,7 @@ class MolGANConvolutionLayer(nn.Module):
     def __init__(self,
                  units: int,
                  nodes: int,
-                 activation=F.tanh,
+                 activation=torch.tanh,
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:3100]: Why is it better to use torch.tanh instead of the tanh from functional?

**Review by rbharath — COMMENTED** [review]
Changes mostly look good but a few minor comments/questions

File: deepchem/models/torch_models/layers.py (hunk 3 above)
Comment (shreyasvinaya) [reply deepchem/models/torch_models/layers.py:3100]: the tanh from functional is deprecated and it raises a warning, so i shifted to torch.tanh as the warning suggested

File: deepchem/models/torch_models/layers.py (hunk 1 above)
Comment (shreyasvinaya) [reply deepchem/models/torch_models/layers.py]: fixed

File: deepchem/models/torch_models/layers.py (hunk 2 above)
Comment (shreyasvinaya) [reply deepchem/models/torch_models/layers.py:3252]: Added

File: deepchem/models/torch_models/tests/test_molgan_layers.py [hunk 4]
```diff
@@ -3,9 +3,8 @@
 try:
     import torch
     # import torch.nn as nn
-    import torch.nn.functional as F
+    # import torch.nn.functional as F
```
Comment (rbharath) [inline deepchem/models/torch_models/tests/test_molgan_layers.py:6]: Can you remove this in the next PR?

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**